modules/aruco/src/aruco.cpp

```diff
@@ -0,0 +1,1573 @@
+                (*idsTmp)[i] = currId;
+            }
+        }
+    }
+
+    private:
+    IdentifyCandidatesParallel &operator=(const IdentifyCandidatesParallel &); // to quiet MSVC
+
+    const Mat *grey;
+    InputArrayOfArrays candidates, contours;
+    const Dictionary *dictionary;
+    vector< int > *idsTmp;
+    vector< char > *validCandidates;
+    DetectorParameters *params;
+};
+
+
+
+/**
+ * @brief Identify square candidates according to a marker dictionary
+ */
+static void _identifyCandidates(InputArray _image, InputArrayOfArrays _candidates,
+                                InputArrayOfArrays _contours, const Dictionary &dictionary,
+                                OutputArrayOfArrays _accepted, OutputArray _ids,
+                                DetectorParameters params,
+                                OutputArrayOfArrays _rejected = noArray()) {
+
+    int ncandidates = (int)_candidates.total();
+
+    vector< Mat > accepted;
+    vector< Mat > rejected;
+    vector< int > ids;
+
+    CV_Assert(_image.getMat().total() != 0);
+
+    Mat grey;
+    _convertToGrey(_image.getMat(), grey);
+
+    vector< int > idsTmp(ncandidates, -1);
+    vector< char > validCandidates(ncandidates, 0);
+
+    //// Analyze each of the candidates
+    // for (int i = 0; i < ncandidates; i++) {
+    //    int currId = i;
+    //    Mat currentCandidate = _candidates.getMat(i);
+    //    if (_identifyOneCandidate(dictionary, grey, currentCandidate, currId, params)) {
+    //        validCandidates[i] = 1;
+    //        idsTmp[i] = currId;
+    //    }
+    //}
+
+    // this is the parallel call for the previous commented loop (result is equivalent)
+    parallel_for_(Range(0, ncandidates),
+                  IdentifyCandidatesParallel(&grey, _candidates, _contours, &dictionary, &idsTmp,
+                                             &validCandidates, &params));
+
+    for(int i = 0; i < ncandidates; i++) {
+        if(validCandidates[i] == 1) {
+            accepted.push_back(_candidates.getMat(i));
+            ids.push_back(idsTmp[i]);
+        } else {
+            rejected.push_back(_candidates.getMat(i));
+        }
+    }
+
+    // parse output
+    _accepted.create((int)accepted.size(), 1, CV_32FC2);
+    for(unsigned int i = 0; i < accepted.size(); i++) {
+        _accepted.create(4, 1, CV_32FC2, i, true);
+        Mat m = _accepted.getMat(i);
+        accepted[i].copyTo(m);
+    }
+
+    _ids.create((int)ids.size(), 1, CV_32SC1);
+    for(unsigned int i = 0; i < ids.size(); i++)
+        _ids.getMat().ptr< int >(0)[i] = ids[i];
+
+    if(_rejected.needed()) {
+        _rejected.create((int)rejected.size(), 1, CV_32FC2);
+        for(unsigned int i = 0; i < rejected.size(); i++) {
+            _rejected.create(4, 1, CV_32FC2, i, true);
+            Mat m = _rejected.getMat(i);
+            rejected[i].copyTo(m);
+        }
+    }
+}
+
+
+/**
+  * @brief Final filter of markers after its identification
```

Comment:
The comment should have why the filtering is done too. This is an obvious comment. Does not add new information.